backends-common/elasticsearch/src/main/java/org/apache/james/backends/es/search/ScrolledSearch.java

```diff
@@ -39,74 +35,74 @@
 
 import com.github.fge.lambdas.Throwing;
 
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+import reactor.core.scheduler.Schedulers;
+
 public class ScrolledSearch {
```

Comment:
you should try to implement ScrolledSearch with Flux.create(sink). It's much easier than using Futures to close the gap between async code and reactor.

Reply:
I looked more complex when I did